.github/workflows/release.yml

```diff
@@ -0,0 +1,103 @@
+name: ci
+
+# This worflow needs this secret:
+#
+# DOCKERPASSWORD = Docker Hub token
+
+on:
+  push:
+    branches: [master]
+    tags: 'v*.*.*'
+  schedule:
+    - cron: "0 13 * * 1"
+
+env:
+  PLATFORMS: "linux/amd64,linux/arm/v7,linux/arm/v6,linux/arm64, linux/ppc64le"     # Build for which platforms
+  IMAGENAME: "ansemjo/speedtest"                                     # Name of the image
+  DEFAULT_TAG: "latest"                                              # Which tag is beeing used if we are building for master/main branch 
+  DOCKER_USER: "ansemjo"                                             # Which user to use to login to DockerHub
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Generate Build-Args
+        id: build-args
+        run: |
+          # echo ::set-output name=build-arg1::"buildarg1"
+          # echo ::set-output name=build-arg2::"buildarg2"
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+      -
+        name: Prepare
+        id: prep
+        run: |
+          if [[ $GITHUB_REF == refs/tags/* ]]; then
+            VERSION=${GITHUB_REF#refs/tags/}
+            if [[ $VERSION =~ ^v([0-9]{1,3})\.([0-9]{1,3})\.([0-9]{1,3})$ ]]; then
+               MAJOR="${BASH_REMATCH[1]}"
+               MINOR="${BASH_REMATCH[2]}"
+               PATCH="${BASH_REMATCH[3]}"
+
+               TAGS="${{ env.IMAGENAME }}:latest"
+               TAGS="${TAGS},${{ env.IMAGENAME }}:${MAJOR}"
+               TAGS="${TAGS},${{ env.IMAGENAME }}:${MAJOR}.${MINOR}"
+               TAGS="${TAGS},${{ env.IMAGENAME }}:${MAJOR}.${MINOR}.${PATCH}"
+            else
+               TAGS="${{ env.IMAGENAME }}:${VERSION}"
+            fi
+          elif [[ $GITHUB_REF == refs/heads/* ]]; then
```

Comment:
What happens if I do push to a branch `devel`? If I read this clause corretly, the image will also be built with the `:latest` tag, no?

Reply:
Nothing happens, as there is only ```master``` specified at top. The whole workflow wouldn't run on other branches. But yes, if you add ```devel``` as well, it would also be tagged as ```latest```.

I personally may opt for using the pull-request workflow for a ```devel``` branch. As it only builds the image to make sure every thin "works" but wouldn't push it.

Reply:
Ah yes, I overlooked that! Then it's clear. :)